### PR TITLE
Support multiple input options in cli

### LINKED ISSUE #502: CLI workflow run command can't process --input param with multiple args
## Expected Behavior
CLI should be able to start a workflow with --input param containing multiple args
See https://docs.temporal.io/docs/learn-cli#start-workflow

## Actual Behavior
`processJSONInput` throws an error when it tries to `Unmarshal` the input. Input param is not a correct json and has to be pre-processed to support the expected behavior.

## Steps to Reproduce the Problem

  1. Execute 
``` bash
./tctl workflow start --tq helloWorldGroup --wt main.WorkflowWith3Args --et 60 -i '"your_input_string" 123 {"Name":"my-string", "Age":12345}'
```
The output will contain an error message `invalid character '1' after top-level value`

## Specifications

  - Version: 0.26.0
  - Platform: Temporal on WSL2/Ubuntu go1.14.4 linux/amd64

### PR DESCRIPTION
**Pull request by feedmeapples**
**What changed?**
Fixes #502 
Added support for multiple input payloads
`tctl workflow start --tq helloWorldGroup --wt main.WorkflowWithInputs --et 60 -i '{"Joker":"Heheheh", "Harley":"Pumpkin!"}' -i '"to infinity and beyond"' -i 'null'`

**Why?**
Before you could specify only one payload as `--input`

**How did you test it?**
Ran the command. Ran tests

**Potential risks**
Low risk. The code shares some logic with --memo flag

### CODE REVIEW

File: tools/cli/flags.go [hunk 1]
```diff
@@ -335,9 +335,9 @@ func getFlagsForStart() []cli.Flag {
 			Usage: "Configure if the same workflow Id is allowed for use in new workflow execution. " +
 				"Options: AllowDuplicate, AllowDuplicateFailedOnly, RejectDuplicate",
 		},
-		cli.StringFlag{
+		cli.StringSliceFlag{
 			Name:  FlagInputWithAlias,
-			Usage: "Optional input for the workflow, in JSON format. If there are multiple parameters, concatenate them and separate by space.",
+			Usage: "Optional input for the workflow, in JSON format. If there are multiple parameters, pass each as a separate input flag.",
```
Comment (mastermanu) [inline tools/cli/flags.go:340]: [Nit] do we need to mention anything about the input flag ordering corresponding to the input on the workflow params, or is this implicit enough for a user to get?

File: tools/cli/util.go [hunk 2]
```diff
@@ -738,51 +738,71 @@ func newContextWithTimeout(c *cli.Context, timeout time.Duration) (context.Conte
...
 func processJSONInput(c *cli.Context) *commonpb.Payloads {
-	rawJson := processJSONInputHelper(c, jsonTypeInput)
-	if len(rawJson) == 0 {
-		return nil
-	}
+	jsonsRaw := readJSONInputs(c, jsonTypeInput)
+
+	var jsons []interface{}
+	for _, jsonRaw := range jsonsRaw {
+		if len(jsonRaw) == 0 {
```
Comment (mastermanu) [inline tools/cli/util.go:745]: whats the scenario for this? passing an intentionally empty argument? Why would we not just ErrorAndExit in this case? Or is this an impossible scenario if invoked from the command line?

File: tools/cli/util.go [hunk 3]
```diff
@@ -738,51 +738,71 @@ func newContextWithTimeout(c *cli.Context, timeout time.Duration) (context.Conte
...
+			ss[0] = fmt.Sprintf("%v", inputsG)
+			inputs = &ss
+		}
+
+		var inputsRaw [][]byte
+		for _, i := range *inputs {
+			inputsRaw = append(inputsRaw, []byte(i))
+		}
+		return inputsRaw
+	} else if c.IsSet(flagInputFileName) {
```
Comment (mastermanu) [inline tools/cli/util.go:797]: is multiple-file input in the next PR?

File: tools/cli/util.go [hunk 4]
```diff
@@ -738,51 +738,71 @@ func newContextWithTimeout(c *cli.Context, timeout time.Duration) (context.Conte
...
-		inputFile := c.String(flagNameOfInputFileName)
+	if c.IsSet(flagRawInput) {
+		inputsG := c.Generic(flagRawInput)
+
+		var inputs *cli.StringSlice
+		var ok bool
+		if inputs, ok = inputsG.(*cli.StringSlice); !ok {
+			// input could be provided as StringFlag instead of StringSliceFlag
+			ss := make(cli.StringSlice, 1)
+			ss[0] = fmt.Sprintf("%v", inputsG)
```
Comment (mastermanu) [inline tools/cli/util.go:788]: this is weird. doesn't this mean its a cli.String type then? Why fmt.Sprintf it?

Comment (feedmeapples) [reply tools/cli/util.go:788]: `cli.String` doesn't exists in the api of urfave/cli. There is only a private field `cli.stringValue`

Comment (feedmeapples) [reply tools/cli/util.go:788]: for reference https://godoc.org/gopkg.in/urfave/cli.v1

Comment (mastermanu) [reply tools/cli/util.go:788]: makes sense

File: tools/cli/flags.go (hunk 1 above)
Comment (alexshtin) [reply tools/cli/flags.go:340]: It's good point. I would add something like "Order of input flags must be the same as parameters order."

File: tools/cli/util.go (hunk 3 above)
Comment (alexshtin) [reply tools/cli/util.go:797]: I hope so.

File: tools/cli/flags.go (hunk 1 above)
Comment (feedmeapples) [reply tools/cli/flags.go:340]: i think they will expect the ordering to matter, i would rather keep the help page relatively simple, focusing on the main information. However we could add this to our docs. I've added an item to Cully regarding the docs

File: tools/cli/util.go (hunk 2 above)
Comment (alexshtin) [reply tools/cli/util.go:745]: How do we pass `nil`? `--input "null"` should work, right? If yes, I would validate this case and `ErrorAndExit("Empty input flag. Pass \"null\" for nil parameter value.")

Comment (feedmeapples) [reply tools/cli/util.go]: using `continue` was indeed not good.
Added support for passing `null`

File: tools/cli/util.go [hunk 5]
```diff
@@ -738,51 +738,78 @@ func newContextWithTimeout(c *cli.Context, timeout time.Duration) (context.Conte
...
+			inputs = &ss
+		}
+
+		var inputsRaw [][]byte
+		for _, i := range *inputs {
+			if i == "nil" {
+				ErrorAndExit("Empty input flag. Pass \"null\" for nil parameter value.", nil)
```
Comment (mastermanu) [inline tools/cli/util.go:795]: Can we add this to the help description up above? Folks that don't use GoLang won't even try to use "nil" and so this error message is going to be hard to discover. Then I would just remove this.

File: tools/cli/util.go [hunk 6]
```diff
@@ -738,51 +738,78 @@ func newContextWithTimeout(c *cli.Context, timeout time.Duration) (context.Conte
...
+			} else if i == "null" {
```
Comment (mastermanu) [inline tools/cli/util.go:796]: [Nit] using string comparison that is case insensitive (https://golang.org/pkg/strings/#example_EqualFold)

File: tools/cli/flags.go (hunk 1 above)
Comment (feedmeapples) [reply tools/cli/flags.go:340]: https://www.notion.so/f03f62f4cefd498fa8caaae2bb834eb4?v=7466982797634db691c9448414f1109a&p=90335cb03b0341319d97160b8281d3ea

**Pull request merged**